platform/security/sts/security-sts-attributequeryclaimshandler/src/main/java/org/codice/ddf/security/claims/attributequery/AttributeQueryClient.java

```diff
@@ -0,0 +1,355 @@
+                reportError(response.getStatus());
+            }
+            return assertion;
+        } catch (IOException | ParserConfigurationException | SAXException e) {
+            throw new AttributeQueryException(
+                    "Unable to parse request string into an XML document.", e);
+        } catch (UnmarshallingException e) {
+            throw new AttributeQueryException("Unable to marshall Element to SAML Response.", e);
+        }
+    }
+
+    /**
+     * If response is unsuccessful, report the cause of the error.
+     *
+     * @param status Status of response object.
+     */
+    private void reportError(Status status) throws AttributeQueryException {
+        switch (status.getStatusCode().getValue()) {
+        case SAML2_UNKNOWN_ATTR_PROFILE:
+            LOGGER.error(
+                    "Unsuccessful response: {}. Incorrect version number or incorrect parsing error.");
+            break;
+        case SAML2_INVALID_ATTR_NAME_VALUE:
+            LOGGER.error("Unsuccessful response: {}. Request attribute name is unknown.");
+            break;
+        case SAML2_UNKNOWN_PRINCIPAL:
+            LOGGER.error(
+                    "Unsuccessful response: {}. Unknown principal name, user is not recognized.");
+            break;
+        default:
+            LOGGER.error(status.getStatusMessage().getMessage());
+            break;
+        }
+        if (status.getStatusMessage() != null && status.getStatusMessage().getMessage() != null) {
+            LOGGER.error(status.getStatusMessage().getMessage());
+        }
+        // Allow bad response to go through.
+    }
+
+    /**
+     * If the connection was successful, continue on, else do not.
+     *
+     * @param statusCode of the connection.
+     */
+    private boolean reportConnectionStatus(int statusCode) {
+        switch (statusCode) {
+        case HttpsURLConnection.HTTP_OK:
+            LOGGER.debug("OK {}. Successful connection to: {}", statusCode,
+                    externalAttributeStoreUrl);
+            return true;
+        case HttpsURLConnection.HTTP_UNAUTHORIZED:
+            LOGGER.warn("Unauthorized {}. Could not connect to: {}", statusCode,
+                    externalAttributeStoreUrl);
+            return false;
+        case HttpURLConnection.HTTP_NOT_FOUND:
+            LOGGER.warn("URL not found {} . Could not connect to: {}", statusCode,
+                    externalAttributeStoreUrl);
+            return false;
+        default:
+            LOGGER.warn("Status code not supported {}.", statusCode);
+            return false;
+        }
+    }
+
+    /**
+     * Creates an Https client for sending the request.
+     *
+     * @param url of the external attribute store.
+     * @return HttpsUrlConnection
+     */
+    protected HttpsURLConnection createHttpsUrlConnection(URL url) {
+        HttpsURLConnection httpsURLConnection = null;
+        try {
+            httpsURLConnection = (HttpsURLConnection) url.openConnection();
```

Comment:
We probably shouldn't be casting this to an HttpsURLConnection unless we're checking that this is actually an HTTPS address.